src/subtypes.ts

```diff
@@ -0,0 +1,2 @@
+export type SubtypeName = string;
+export const isValidSubtypeName = (s: any, t: any): s is SubtypeName => true; // Placeholder before further commits
```

Comment:
This is just a dummy function rn, can you instead implement this or lets remove it unless we come up with the  usage

Reply:
I added the implementation from #2993 in 2a73ed1.